swingset/src/main/java/com/nqadmin/swingset/datasources/RowSetOps.java

```diff
@@ -669,9 +733,119 @@ public static Class<?> findJavaTypeClass(final JDBCType _jdbcType)
 			case LONGNVARCHAR:
 				clazz = String.class;
 				break;
+
+			// case ARRAY:
+			// 	clazz = java.sql.Array.class;
+			// 	break;
+
+			// case BINARY:
+			// case VARBINARY:
+			// case LONGVARBINARY:
+			// 	clazz = byte[].class;
+			// 	break;
+
+			// case CLOB: clazz = java.sql.Clob.class; break;
+			// case BLOB: clazz = java.sql.Blob.class; break;
+			// case REF: clazz = java.sql.Ref.class; break;
+			// case DATALINK: clazz = java.net.URL.class; break;
+			// case ROWID: clazz = java.sql.RowId.class; break;
+			// case NCLOB: clazz = java.sql.NClob.class; break;
+			// case SQLXML: clazz = java.sql.SQLXML.class; break;
 			default:
+				// TODO: SSSQLExceptionUnhandledType
 				throw new SQLException("Unhandled type: " + _jdbcType);
 		}
 		return clazz;
+
+		// case DISTINCT: Object type of underlying type
+		// case STRUCT: java.sql.Struct or java.sql.SQLData
+		// case JAVA_OBJECT: Underlying Java class
+	}
+
+	// FOLLOWING ONLY USED FROM SSTableModel (AT LEAST FOR NOW)
+
+
+	/**
+	 * Update the Grid's RowSet at the specified column index with the given Object value.
+	 * RowSet. Operate on the current row.
+	 * <p>
+	 * When the user changes/edits the SSDataGrid cell this method propagates the
+	 * change to the RowSet. A separate call is required to flush/commit the change
+	 * to the database.
+	 *
+	 * @param _rowSet RowSet on which to operate
+	 * @param _value string to be type-converted as needed and updated in
+	 *                      underlying RowSet column
+	 * @param _columnIndex   index of the database column
+	 * @param type NOT USED, the jdbc driver does the conversion
+	 * @throws SQLException  thrown if a database error is encountered
+	 */
+	public static void updateColumnObject(RowSet _rowSet, Object _value, int _columnIndex, JDBCType type) throws SQLException {
+		_rowSet.updateObject(_columnIndex, _value);
+	}
+
+	/**
+	 * Update the Grid's RowSet at the specified column index with the given Object value.
+	 * RowSet. Operate on the current row.
+	 * <p>
+	 * When the user changes/edits the SSDataGrid cell this method propagates the
+	 * change to the RowSet. A separate call is required to flush/commit the change
+	 * to the database.
+	 *
+	 * @param _rowSet RowSet on which to operate
+	 * @param _value string to be type-converted as needed and updated in
+	 *                      underlying RowSet column
+	 * @param _columnIndex   index of the database column
+	 * @param type The JDBCType of the column
+	 * @throws SQLException  thrown if a database error is encountered
+	 * @see <a href="https://download.oracle.com/otn-pub/jcp/jdbc-4_3-mrel3-eval-spec/jdbc4.3-fr-spec.pdf">JDBC 4.3 Specification</a> Appendix B
+	 */
+	public static void updateColumnObject2(RowSet _rowSet, Object _value, int _columnIndex, JDBCType type) throws SQLException {
```

Comment:
Can we get the JDBCType by calling the getJDBCColumnType function rather than getting JDBCType as argument?

Reply:
I thought, but could be very wrong, that accessing meta data was very expensive.

I'll make the two changes discussed below.

I see that `updateColumnObject()` is used in 3 places. Could have an alternate version of `updateColumnObject()` that only takes two arguments which means that the type should be figured out automatically. That would be best because one version assumes doesn't need the type info.

Notice that `updateColumnObject()` and `updateColumnObject2()` take the same args. One depends on `updateObject` working, the other is a fallback; since the object type is correct it problably works, but the alternate is available. One thought is to have xxx1() and xxx2 both, and have `updateColumnObject()` calls one or the other, making it even simpler.

Reply:
DONE

Reply:
I was more thinking of just having one function with below signature. Then the first line of this function would be JDBCType  type = RowSetOps.getJDBCColumnType(rowset, _column + 1);
The switch block would follow this line.

public static void updateColumnObject(RowSet _rowSet, Object _value, int _columnIndex) throws SQLException {

Line 850 of SSTableModel will be updated to remove the type argument. Which is the only place where we are passing type argument to updateColumnObject.

Reply:
I understand. I have a tendency to over-optimize. The function you mention
```
public static void updateColumnObject(
    RowSet _rowSet, Object _value, int _columnIndex) throws SQLException {
```
exists. The other funtion
```
public static void updateColumnObject(
    RowSet _rowSet, Object _value, int _columnIndex, JDBCType type) throws SQLException {
```
is available if you already have the type so you can avoid accessing meta data. (I thought that was expensive).

And in some sense, I see the stuff in RowSetOps in a transitional state, since I'm not sure what works with any give jdbc driver. I'm hoping that we can get rid of the type entirely since we've already got an object of the expected type. Notice that the way it's set up now
```
RowSetOps.getJDBCColumnType(rowset, _column + 1);
```
is never executed since
```
_rowSet.updateObject(_columnIndex, _value);
```
doesn't take a type.